hypnagogic_core/src/util/icon_ops.rs

```diff
@@ -0,0 +1,45 @@
+use dmi::icon::IconState;
+
+// Removes duplicate frames from the icon state's animation, if it has any
+pub fn dedupe_frames(mut icon_state: IconState) -> IconState {
+    if icon_state.frames <= 1 {
+        return icon_state;
+    }
+    let current_delays = match icon_state.delay {
+        Some(delay) => delay,
+        None => return icon_state,
+    };
```

Comment:
Good candidate for the fairly recently stablized let else
```suggestion
    let Some(current_delays) = icon_state.delay else {
        return icon_state;
    };
```